cloudbot/util/decorators.py

```diff
@@ -0,0 +1,25 @@
+from time import time
+
+SPAM_LIST = {}
+
+
+def spam_guard(func):
+    def wrapper(*args, **kwargs):
+        nick = kwargs.get('nick', None)  # pull the nick from the args
+        if nick:  # if there is no nick specified for whatever reason, carry on
+            global SPAM_LIST
+            if SPAM_LIST.get(nick, None):
+                SPAM_LIST[nick][0] = time()
+                if time() - SPAM_LIST[nick][0] < SPAM_LIST[nick][1]:
+                    # notice that they are spamming
+                    SPAM_LIST[nick][1] += 10
+                    return (
+                        "{} attempted to execute a command within their previous" \
+                        " limit. An additional 10 seconds has been added. Current" \
+                        " limit is {} seconds.".format(nick, SPAM_LIST[nick][1])
```

Comment:
1. The return notifies the channel. This has two effects. It notifies the user and it reminds everyone else the punishment for spamming.
2. There is no problem with this. The reloads will not be on a per-user basis. But most likely a monthly refresh. Or whenever the OPs feel a refresh is warranted. As far as the saving. I want to have this be done automatically by the bot on a regular interval. (a future improvement).
3. This is the point.
4. And that is the beauty of using this system. We can choose what functions will be decorated with the spam guard. It is completely up to us.